Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -47,15 +46,13 @@ public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto)
             var saved = await _unitOfWork.SaveAsync();
             if (!saved)
             {
-                if(await _jokesRepo.GetJokeDtoAsync(joke.Id) == null)
-                {
-                    throw new AppException("Error with saving the Joke.");
-                }
+                throw new AppException("Error with saving the Joke.");
             }
 
-            var jokeReplyDto = _mapper.Map<JokeReplyDto>(joke);
+            var jokeDto = _mapper.Map<JokeDto>(joke);
+            jokeDto.AuthorName = user.Name;
```

Comment:
For me this doesn't "feel right"...

Reply:
This is fine for now. there is a second way.
GetUserReplyAsync then you don't get all fields from the user.

Reply:
> use AddAuthorToJoke(jokeDto);

I already have the user in the scope of this method. In the method "AddAuthorToJoke(jokeDto)" it's retrieved again by Id. It would be an unnecessary call to the UserManager.